include/ros2_logging_fmt/ros2_logging_fmt.hpp

```diff
@@ -15,49 +15,113 @@ class Logger
   Logger(rclcpp::Logger rclcpp_logger): rclcpp_logger_(rclcpp_logger)
   {}
 
+  Logger(rclcpp::Logger rclcpp_logger, rclcpp::Clock::SharedPtr clock): rclcpp_logger_(rclcpp_logger)
+  { clk = clock; }
+
   template<typename... Args>
   void info(const char* str, Args... args)
   {
     RCLCPP_INFO(rclcpp_logger_, write_buffer(str, args...) );
   }
 
+  template<typename... Args>
+  void info_throttle(const char* str, rclcpp::Clock::SharedPtr clock, int duration, Args... args)
+  {
+      RCLCPP_INFO_THROTTLE(rclcpp_logger_, *clock, duration, write_buffer(str, args...) );
```

Comment:
As you are always dereferencing the clock object, why take a shared_ptr to the clock object and not a const& ?

Reply:
My C++ is weak here, and you make a good point.

Reply:
Moved this to a clock reference and working.